invenio_indexer/cli.py

```diff
@@ -52,3 +53,19 @@ def run(delayed, concurrency):
     else:
         click.secho("Indexing records...", fg='green')
         RecordIndexer().process_bulk_queue()
+
+
+@index.command()
+@click.argument('pid_type')
+@click.option('-i', '--index', default=None, help="the index name.")
+@click.option('-d', '--doctype', default=None, help="the doc_type.")
+@with_appcontext
+def reindex(pid_type, index, doctype):
+    """Reindex all records."""
+    query = (x[0] for x in PersistentIdentifier.query.filter_by(
+        pid_type=pid_type, object_type='rec'
+    ).values(
+        PersistentIdentifier.object_uuid
+    ))
+    click.echo("Sending tasks...")
+    RecordIndexer().bulk_index(query, index=index, doc_type=doctype)
```

Comment:
As I know  you should call `process_bulk_queue()` for syn call or `process_bulk_queue.delay()` for async call. Here is what I'm using actually, it is probably not optimal, but i t works for me:

        @utils.command()
        @click.argument('pid_type')
        @click.option('--bulk-size', '-s', default=50)
        @with_appcontext
        def reindex(pid_type, bulk_size):
            """Reindex using bulks."""
            query = (x[0] for x in PersistentIdentifier.query.filter_by(
                pid_type=pid_type, object_type='rec'
            ).values(
                PersistentIdentifier.object_uuid
            ))
            click.echo("Sending tasks...")

            from invenio_indexer.tasks import index_record, process_bulk_queue
            #for rec in query:
                #index_record.delay(str(rec))
                #click.echo(rec)
            to_index = []
            for rec in query:
                to_index.append(rec)
                if len(to_index) == bulk_size:
                    RecordIndexer().bulk_index(to_index)
                    process_bulk_queue.delay()
                    to_index = []
                    
            if to_index:
                click.echo("Last %d recds" % len(to_index))
                RecordIndexer().bulk_index(to_index)
                process_bulk_queue.delay()